### DISCUSSION TITLE
Using JSONC

### DISCUSSION THREAD

**Discussion by gabrielomana** [fastfetch-cli/fastfetch#498] [Q&A]
Hello everyone,

First of all, I apologize, because I really don't have much knowledge of these topics, but could someone please explain to me how to use .jsonc files with Fastfetch? that is, where should I save the .jsonc files? How can I test them? How do I make the customization persistent?

I have Debian 12, and when I install fasfetch, no directory is created in ~/.config/, I only have the config.conf from /etc/fastfetch

I think this information should be detailed in the README of the project.

Thanks and regards,

**Comment by CarterLi** [comment]
The support of JSONC is not in stable release yet. You must install [the dev build](https://github.com/fastfetch-cli/fastfetch/actions)

A JSONC config file should be located in `~/.config/fastfetch`, namely `config.jsonc`. You must create it yourself. You may start it with

```jsonc
// ~/.config/fastfetch/config.jsonc
{
    "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json"
}
```

Using an editor with JSON schema support is highly suggested. If unsure, I suggest using [vscode](https://code.visualstudio.com/).

<img width="628" alt="image" src="https://github.com/fastfetch-cli/fastfetch/assets/6134068/60159686-9b9f-4500-b87b-b5181279f7f6">

> I have Debian 12, and when I install fasfetch, no directory is created in ~/.config/, I only have the config.conf from /etc/fastfetch

We don't create config files automatically. You may create it with `fastfetch --gen-config-file`, which creates a traditional flag based `.conf` file. Creating a `.jsonc` is not supported.

> I think this information should be detailed in the README of the project.

Yes
